src/components/CalendarMonth.jsx

```diff
@@ -104,7 +107,7 @@ export default class CalendarMonth extends React.Component {
     } = this.props;
 
     const { weeks } = this.state;
-    const monthTitle = month.format(monthFormat);
+    const monthTitle = renderMonth ? renderMonth(month) : month.format(monthFormat);
```

Comment:
If we're calling a user function here, we'd need to somehow assert that `monthTitle` is a string, to match `month.format()` - and throw if not.